crates/outline_panel/src/outline_panel.rs

```diff
@@ -4093,6 +4156,40 @@ impl Render for OutlinePanel {
     }
 }
 
+fn find_active_indent_guide_ix(
+    outline_panel: &OutlinePanel,
+    selected_entry: &PanelEntry,
+    candidates: &[IndentGuideLayout],
+) -> Option<usize> {
+    let (target_ix, target_depth) = outline_panel
+        .cached_entries
+        .iter()
+        .enumerate()
+        .find(|(_, cached_entry)| &cached_entry.entry == selected_entry)
+        .map(|(i, entry)| (i, entry.depth))?;
+
+    let (target_ix, target_depth) = if let Some(target_depth) = outline_panel
```

Comment:
We could cache `target_ix` and `target_depth` on `SelectedEntry::Valid` and only re-compute it each time we call `select_entry` in order to avoid iterating all the cached entries on each render.